hotel_restaurant/models/hotel_restaurant.py

```diff
@@ -0,0 +1,690 @@
+                "folio_id": record.folio_id.id,
+                "table_nos_ids": [(6, 0, table_ids)],
+                "is_folio": record.is_folio,
+            }
+            reservation_order.create(values)
+        self.write({"state": "order"})
+        return True
+
+    @api.onchange("customer_id")
+    def _onchange_partner_id(self):
+        """
+        When Customer name is changed respective adress will display
+        in Adress field
+        @param self: object pointer
+        """
+        if not self.customer_id:
+            self.partner_address_id = False
+        else:
+            addr = self.customer_id.address_get(["default"])
+            self.partner_address_id = addr["default"]
+
+    @api.onchange("folio_id")
+    def _onchange_folio_id(self):
+        """
+        When you change folio_id, based on that it will update
+        the customer_id and room_number as well
+        ---------------------------------------------------------
+        @param self: object pointer
+        """
+        for rec in self:
+            if rec.folio_id:
+                rec.write(
+                    {
+                        "customer_id": rec.folio_id.partner_id.id,
+                        "room_id": rec.folio_id.room_line_ids[0].product_id.id,
+                    }
+                )
+
+    def action_set_to_draft(self):
+        """
+        This method is used to change the state
+        to draft of the hotel restaurant reservation
+        --------------------------------------------
+        @param self: object pointer
+        """
+        self.write({"state": "draft"})
+
+    def table_reserved(self):
+        """
+        when CONFIRM BUTTON is clicked this method is called for
+        table reservation
+        @param self: The object pointer
+        @return: change a state depending on the condition
+        """
+        for reservation in self:
+            reservation._cr.execute(
+                "select count(*) from "
+                "hotel_restaurant_reservation as hrr "
+                "inner join reservation_table as rt on \
+                             rt.reservation_table_id = hrr.id "
+                "where (start_date,end_date)overlaps\
+                             ( timestamp %s , timestamp %s ) "
+                "and hrr.id<> %s and state != 'done'"
+                "and rt.name in (select rt.name from \
+                             hotel_restaurant_reservation as hrr "
+                "inner join reservation_table as rt on \
+                             rt.reservation_table_id = hrr.id "
+                "where hrr.id= %s) ",
+                (
+                    reservation.start_date,
+                    reservation.end_date,
+                    reservation.id,
+                    reservation.id,
+                ),
+            )
+            res = self._cr.fetchone()
+            roomcount = res and res[0] or 0.0
+            if not reservation.table_nos_ids:
```

Comment:
Move validation at the top so we can exclude executing sql in case of validation